### ISSUE TITLE
Amazon UK 2009 displays no results

### ISSUE THREAD

**Issue by danniboy** [philipmulcahy/azad#155]
I have literally run every year on the account but 2009 for Amazon UK never displays the report on the screen.

Attached is the log

In running all these reports I have noticed that the total on the screen report is incorrect. Should I submit another issue?

[www.amazon_2009.log](https://github.com/philipmulcahy/azad/files/6216464/www.amazon_2009.log)

**Comment by philipmulcahy** [comment]
I am guessing that Amazon have changed something on the 2009 data that is triggering a bug.
The log is not that informative. Have you worked through the self help steps in README.md?

**Comment by philipmulcahy** [comment]
On the totals, I think a seperate bug is appropriate.
Please include marked up screenshots, the matching csv and a description of the figure you think is wrong and what it should be.
Thanks!

**Comment by danniboy** [comment]
I cleared the site data and I have also regularly cleared the cache from the extension itself.

Is there anything else I could do?

See screenshots of the extension while running 2009

![Screenshot 2021-03-29 at 11 36 25](https://user-images.githubusercontent.com/201021/112831302-46ed3b80-9083-11eb-8df1-973cad6a9011.png)
![Screenshot 2021-03-29 at 11 37 06](https://user-images.githubusercontent.com/201021/112831308-481e6880-9083-11eb-9217-284986924d9a.png)

**Comment by philipmulcahy** [comment]
I am working on the hypothesis that there is one or more "bad" orders and a shortsighted bit of my code blows up the table drawing for all of the orders because it cannot handle the bad one.
I plan (when I get time) to modify my code to log the bad ones and get on and display the good ones.
We can them home in on what's happening. ETA 1w.

**Comment by philipmulcahy** [comment]
Here's an experimental version that _might_ get further, and tell give us a clue about what is causing gthe problem: https://www.mulcahyfamily.org/azad/azad_f1521c62.zip
To use, please read: https://github.com/philipmulcahy/azad/blob/master/README.md#how-to-try-out-an-experimental-version.
I'd be grateful for a console log from this one...
Yours,
Philip

**Comment by danniboy** [comment]
Here you go
[www.amazon_2009_2.log](https://github.com/philipmulcahy/azad/files/6254898/www.amazon_2009_2.log)

**Comment by philipmulcahy** [comment]
OK - here's an "even better" one - same again please!
https://www.mulcahyfamily.org/azad/azad_0a7d44af.zip

**Comment by danniboy** [comment]
[www.amazon_2009_3.log](https://github.com/philipmulcahy/azad/files/6267616/www.amazon_2009_3.log)

**Comment by philipmulcahy** [comment]
Thank you!

Here's a new dev build: www.mulcahyfamily.org/azad/azad_3ddf0e15.zip

Possible outcomes:
a) it works -> let me know and I'll release it properly in the next few days.
b) it works, but one or more orders is missing from the output -> send me a log again please.
c) no change -> log please!

Yours,

Philip

**Comment by danniboy** [comment]
This was strange. I logged out and back in to Amazon UK and ran the report and nothing showed up then I opened the console to get the log for you and I cleared the console and ran again and then it worked.

The report has 60 orders which matches what Amazon says **60 orders placed in 2009**. I cannot confirm if the amounts are correct.

There was one order that had 45 charges. Could that have been the issue?

Anyway here is the log
[www.amazon_2009_4.log](https://github.com/philipmulcahy/azad/files/6289341/www.amazon_2009_4.log)
